### ISSUE TITLE
discrepancy between coalescence rates for msprime and stdpopsim OOA model

### ISSUE THREAD

**Issue by petrelharp** [tskit-dev/msprime#1618]
This is probably a bug in how we compute coalescence rates with the new demographic models, but could conceivably be a bug in `msprime._ooa_model()`:
```
import msprime
import stdpopsim
import numpy as np

msp = msprime.Demography._ooa_model().debug()

species = stdpopsim.get_species("HomSap")
model = species.get_demographic_model("OutOfAfrica_3G09")
sps = msprime.Demography.from_old_style(
        population_configurations=model.population_configurations,
        migration_matrix=model.migration_matrix,
        demographic_events=model.demographic_events).debug()

T = np.concatenate([
            np.linspace(0, 1000, 2001),
            np.linspace(1000, 1e4, 401)[1:]
])

msp_R, _ = msp.coalescence_rate_trajectory(T, [0, 2, 0, 0, 0, 0])
sps_R, _ = sps.coalescence_rate_trajectory(T, [0, 2, 0])

print("msprime:", 1/(2*msp_R[-3:]))
print("stdpopsim:", 1/(2*sps_R[-3:]))
```
produces:
```
msprime: [-5.37590936e+257 -3.17744309e+267 -1.87803475e+277]
stdpopsim: [7300. 7300. 7300.]
```
These should be identical. *Note:* there's a warning about possible numerical instability, but that's not what's going on here.

*Edit:* actually using the current msprime, I get
```
msprime: [31995.80756988 32072.04220953 32148.51218145]
stdpopsim: [7300. 7300. 7300.]
```
which is much better but still not identical.

**Comment by petrelharp** [comment]
The problem with OOA is fixed in #1620 :tada:; so this now depends on that.

**Comment by jeromekelleher** [comment]
Closed in #1620
